src/credentials/GroupDataProvider.h

```diff
@@ -227,8 +228,8 @@ class GroupDataProvider
      *  @retval #CHIP_ERROR_INCORRECT_STATE if called when already initialized.
      *  @retval #CHIP_NO_ERROR on success
      */
-    virtual CHIP_ERROR Init() = 0;
-    virtual void Finish()     = 0;
+    virtual CHIP_ERROR Init(PersistentStorageDelegate * storage) = 0;
```

Comment:
This init should not rely on `PersistentStorageDelegate`.

Suggest that we keep init no-params here, and have `GroupDataProviderImpl::SetPersistentStorageDelegate(PersistentStorageDelegate *)`.

Then `GroupDataProviderImpl::Init()`  could just fail with CHIP_ERROR_INCORRECT_STATE if `SetPersistentStorageDelegate` not called.

Reply:
I have removed the parameter from the `Init` method, and replaced with a new `SetStorageDelegate`.